src/phases/stat-stage-change-phase.ts

```diff
@@ -16,60 +16,73 @@ import { getStatKey, getStatStageChangeDescriptionKey, Stat, type BattleStat } f
 import i18next from "i18next";
 import { settings } from "#app/system/settings/settings-manager";
 import { PokemonPhase } from "./abstract-pokemon-phase";
+import { ReflectStatStageChangeAbAttr } from "#app/data/ab-attrs/reflect-stat-stage-change-ab-attr";
 
 export type StatStageChangeCallback = (changed: BattleStat[], relativeChanges: number[], target?: Pokemon) => void;
 
 interface SSCPhaseOptions {
   showMessage?: boolean;
   ignoreAbilities?: boolean;
   canBeCopied?: boolean;
+  bypassReflect?: boolean;
   onChange?: StatStageChangeCallback;
 }
 
 export class StatStageChangePhase extends PokemonPhase {
   protected readonly stats: BattleStat[];
-  protected readonly selfTarget: boolean;
+  protected readonly source: Pokemon | null;
   protected stages: number;
   protected readonly showMessage: boolean;
   protected readonly ignoreAbilities: boolean;
   protected readonly canBeCopied: boolean;
+  protected readonly bypassReflect: boolean;
   protected readonly onChange?: StatStageChangeCallback;
   private readonly options?: SSCPhaseOptions;
 
   constructor(
     battlerIndex: BattlerIndex,
-    selfTarget: boolean,
+    source: Pokemon | null,
     stats: BattleStat[],
     stages: number,
     options?: SSCPhaseOptions,
   ) {
     super(battlerIndex);
 
-    this.selfTarget = selfTarget;
+    this.source = source;
     this.stats = stats;
     this.stages = stages;
     this.showMessage = options?.showMessage ?? true;
     this.ignoreAbilities = options?.ignoreAbilities ?? false;
     this.canBeCopied = options?.canBeCopied ?? true;
+    this.bypassReflect = options?.bypassReflect ?? false;
     this.onChange = options?.onChange;
     this.options = options;
   }
 
   public override start(): void {
     const pokemon = this.getPokemon();
+    const selfTarget = pokemon === this.source;
```

Comment:
```suggestion
    const selfTarget = pokemon.id === this.source?.id;
```
would it be safer to check for the id?

Reply:
I don't think it makes much of a difference? Comparing objects with `===` just compares their references, and I don't think comparing `id` instead of the reference changes much